packages/atomic/CONTRIBUTING.md

```diff
@@ -0,0 +1,255 @@
+
+export const Default: Story = {};
+
+export const WithCustomIcons: Story = {
+  name: 'With custom icons',
+  args: {
+    'attributes-previous-button-icon': '...',
+    'attributes-next-button-icon': '...',
+  },
+};
+```
+
+### Documentation (.mdx)
+
+Comprehensive component documentation with examples and usage instructions.
+
+**Key structure:**
+- **Must use the `AtomicDocTemplate` Storybook block** - This provides consistent documentation structure and integrates with Storybook stories
+- **Usage section as children** - Component-specific usage instructions and examples are provided as children of the template
+- The template automatically handles property/attribute reference, event documentation, and API integration
+
+**Example structure:**
+```typescript
+import { Meta } from '@storybook/blocks';
+import * as AtomicMyComponentStories from './atomic-my-component.new.stories';
+import { AtomicDocTemplate } from '../../../../storybook-utils/documentation/atomic-doc-template';
+
+<Meta of={AtomicMyComponentStories} />
+
+<AtomicDocTemplate
+  stories={AtomicMyComponentStories}
+  githubPath="commerce/atomic-my-component/atomic-my-component.ts"
+  tagName="atomic-my-component"
+  className="AtomicMyComponent"
+>
+This component provides [component description and purpose].
+
+**Note:** [Any important usage notes or constraints]
+
+```html
+<atomic-commerce-interface>
+  ...
+  <atomic-commerce-layout>
+    ...
+    <atomic-layout-section section="main">
+      
+      <atomic-my-component></atomic-my-component>
+
+    </atomic-layout-section>
+  </atomic-commerce-layout>
+</atomic-commerce-interface>
+```
+
+</AtomicDocTemplate>
+```
+
+**Required content within the template for the Usage section:**
+- Component description and purpose
+- Basic usage examples with proper HTML structure
+- Any important notes or constraints
+- Integration context within the atomic interface
+
+### Unit Tests (.spec.ts)
+
+Comprehensive unit testing for component logic and behavior.
+
+**Testing principles:**
+- Follow the [Atomic test conventions](https://coveord.atlassian.net/wiki/spaces/JSUI/pages/5221122259/Atomic+test+conventions)
+- Wrap all tests under a single `describe` with the component name
+- Use `it` function (not `test`) for individual test cases
+- Start test descriptions with "should"
+
+**Required test coverage:**
+- **Test all possible functionalities** - Every prop and public method of the component must be tested
+- **Test all condition paths** - Cover all if/else branches, switch cases, and conditional logic
+- **Test all DOM attributes** - Validate proper rendering of ARIA labels, CSS parts, and other DOM properties
+- **Test component states** - Verify component behavior in different states (loading, error, success, etc.)
+- **Test user interactions** - Cover all click handlers, keyboard navigation, and event listeners
+- **Test property/attribute binding** - Ensure all component properties correctly affect the rendered output
+
```

Comment:
We should recommend using the prompt to generate the bulk of the tests for a new or migrated component!

We'll also have prompts for generating the core stories and end-to-end tests, so we should recommend using these as well.